### PR TITLE
FavoritesVC with Combine

### PR DESCRIPTION
**Pull request by lihter**
- Implemented UICollectionViewDiffableDataSource to FavoritesVC and it's CV
- Conformed MovieViewModel to Hashable
- Made functions (computed properties) in MovieRepo, MovieUC and FavoritesPresenter for fetching favorite movies

https://user-images.githubusercontent.com/82365531/130081577-67817e43-b3f6-4d9a-b9d0-1d8e34486aba.mp4

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
 
 class FavoritesViewController: UIViewController {
     
     let offset: CGFloat = 4
-    
-    var movies: [MovieViewModel]!
+        
+    var disposables = Set<AnyCancellable>()
+    typealias DataSource = UICollectionViewDiffableDataSource<FavoritesSection, MovieViewModel>
+    typealias Snapshot = NSDiffableDataSourceSnapshot<FavoritesSection, MovieViewModel>
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift:7]: typealias goes on top, than let, then var, private let, private var

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift [hunk 2]
```diff
@@ -1,10 +1,14 @@
+import Combine
 import UIKit
 
 class FavoritesViewController: UIViewController {
     
     let offset: CGFloat = 4
-    
-    var movies: [MovieViewModel]!
+        
+    var disposables = Set<AnyCancellable>()
```
Comment (fban1) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift]: this can be private

**Review by fban1 — APPROVED** [review]

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift (hunk 2 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift]: [Fixed](https://github.com/lihter/Movie-App/pull/47/commits/806b235c6a5de094cad61479512ac2b4c2787c53#diff-1f19f68183b9d8b0e617e1c1d06849efdf8b40b09f513bae95cfc63863e4f5d6R17)

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController.swift:7]: [Fixed](https://github.com/lihter/Movie-App/pull/47/commits/806b235c6a5de094cad61479512ac2b4c2787c53#diff-1f19f68183b9d8b0e617e1c1d06849efdf8b40b09f513bae95cfc63863e4f5d6R6-R7)

**Review by fban1 — APPROVED** [review]

**Pull request merged**